les/client_handler.go

```diff
@@ -20,416 +20,509 @@ import (
 	"context"
 	"math/big"
 	"math/rand"
-	"sync"
-	"sync/atomic"
-	"time"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/mclock"
 	"github.com/ethereum/go-ethereum/core/forkid"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/eth/protocols/eth"
 	"github.com/ethereum/go-ethereum/les/downloader"
+	"github.com/ethereum/go-ethereum/les/flowcontrol"
+	vfc "github.com/ethereum/go-ethereum/les/vflux/client"
 	"github.com/ethereum/go-ethereum/light"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/p2p"
-	"github.com/ethereum/go-ethereum/params"
 )
 
 // clientHandler is responsible for receiving and processing all incoming server
 // responses.
 type clientHandler struct {
-	ulc        *ulc
 	forkFilter forkid.Filter
-	checkpoint *params.TrustedCheckpoint
-	fetcher    *lightFetcher
-	downloader *downloader.Downloader
-	backend    *LightEthereum
-
-	closeCh chan struct{}
-	wg      sync.WaitGroup // WaitGroup used to track all connected peers.
+	blockchain *light.LightChain
+	peers      *peerSet
+	retriever  *retrieveManager
+
+	// only for PoW mode
+	fetcher        *lightFetcher
+	downloader     *downloader.Downloader
+	noInitAnnounce bool
+}
 
-	// Hooks used in the testing
-	syncStart func(header *types.Header) // Hook called when the syncing is started
-	syncEnd   func(header *types.Header) // Hook called when the syncing is done
+// sendHandshake implements handshakeModule
+func (h *clientHandler) sendHandshake(p *peer, send *keyValueList) {
+	sendGeneralInfo(p, send, h.blockchain.Genesis().Hash(), forkid.NewID(h.blockchain.Config(), h.blockchain.Genesis().Hash(), h.blockchain.CurrentHeader().Number.Uint64()))
+	p.announceType = announceTypeSimple
+	send.add("announceType", p.announceType)
+	sendHeadInfo(send, blockInfo{})
```

Comment:
Please add a comment that: 

the chain head information of clients are not needed by servers, so pass null here.